### DISCUSSION TITLE
osu!mania-styled scroll speed changes in osu!taiko

### DISCUSSION THREAD

**Discussion by Hiviexd** [ppy/osu#27350] [Ideas]
having mania-styled SVs has been a distant dream for tons of osu!taiko mappers mainly because they provide a heightened playing skill ceiling alongside the vast amount of new ways to represent a song (choppy bursts, linear/exponential playfield speed changes, abrupt pauses, etc etc).

i've been dabbling into lazer dev and discovered that having this exist is extremely easy by just changing `DrawableTaikoRuleset`'s `VisualisationMethod` to `Sequential`, and it would be just a matter of mappers doing their magic with this.

i've made a rough proof of concept on how this could look like in gameplay

https://github.com/ppy/osu/assets/62819481/867475e3-5249-41af-a68b-27bad1a1c98f

this was made by just copying the ranked osu!mania beatmap SVs of this song onto the other ranked osu!taiko map of this, so things would look way better if the SV changes were altered to better suit the taiko map.

i've been passing this video around taiko mappers and the reactions were very positive to say the least, so i'd love is something like this could be officially implemented, and i honestly imagine that this would be extremely revolutionary to the mapping meta.

ideally we'd have this as a beatmap setting that mappers control (switch between `Sequential` and `Overlapping` visualisation methods), but someone suggested to me to have this as a "fun" mod, so it'd be a novelty on current maps, while giving opportunity to mappers to SV around this mod. not sure on how viable/future-proof this approach would be.

**Comment by Alchyr** [comment]
if I had to say what I *really* want, it'd be to be able to use both in a single map

but that would require much more significant changes to how stuff is setup, and so isn't really a feasible thing to add (as far as I'm aware)

**Reply by Hiviexd** [reply]
with the direction of lazer favoring per-object SVs over the traditional timing point approach, i doubt having both option would be reasonably feasible, so i opted to suggest the next best thing which is to at least have one of the two.

**Comment by TheFearlessDeath500** [comment]
this could be similar like HBSCROLL or BMSCROLL in taiko simulators no?

**Comment by peppy** [comment]
Would you see this being global to the beatmap? ie. no ability to mix-and-match in the same map the two modes? If so, I could see this being feasible.

**Reply by Hiviexd** [reply]
that would be a good start (and the original intention of the post) yes
